src/rest.wiki

```diff
@@ -277,21 +280,27 @@ let edit_handler_aux ?(create = false) id_opt (content_type, raw_content_opt) =
     | _, None ->
       send_error ~code:400 "Body content is missing"
     | Some id, Some raw_content ->
-      read_raw_content raw_content ~>>= fun location_str ->
+      read_raw_content raw_content >>= fun location_str ->
       catch (fun () ->
           (if create then
             Lwt.return_unit
-          else
-            Ocsipersist.find db id ~>>= fun _ -> Lwt.return_unit)
-          ~>>= fun () ->
-          let location = Yojson.from_string<location> location_str in
-          Ocsipersist.add db id location ~>>= fun () ->
+           else
+            Ocsipersist.find db id >>= fun _ -> Lwt.return_unit)
+          >>= fun () ->
+          let location =
+            (let open Result in
+            let loc_result = location_of_yojson (Yojson.Safe.from_string location_str) in
+            (function
+              Ok loc-> loc
+            | Error _ -> raise Deriving_Yojson.Failed ) loc_result)  in
+          Ocsipersist.add db  id location >>= fun () ->
           send_success ())
       (function
         | Not_found ->
           send_error ~code:404 ("Location not found: " ^ id)
         | Deriving_Yojson.Failed ->
-          send_error ~code:400 "Provided JSON is not valid")
+          send_error ~code:400 "Provided JSON is not valid"
+        | _ -> assert false)
```

Comment:
I am not sure what will happen here. Probably an internal server error? In any case, you can explicitly do something like `send_error ~code:500 "Internal server error"`